### PR TITLE
feat: Support Copy

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
fixes https://github.com/oras-project/oras-dotnet/issues/5

### CODE REVIEW

File: Oras.Tests/MemoryTest/MemoryTargetTest.cs [hunk 1]
```diff
@@ -231,5 +231,71 @@ public async Task MemoryTarget_ShouldReturnPredecessorsOfNodes()
...
+            await sourceTarget.TagAsync(root, reference, cancellationToken);
+            var destinationTarget = new MemoryTarget();
+            var gotDesc = await Copy.CopyAsync(sourceTarget, reference, destinationTarget, "", cancellationToken);
+            Assert.Equal(gotDesc, root);
+
+            foreach (var des in descs)
+            {
+                await destinationTarget.ExistsAsync(des, cancellationToken);
+            }
+        }
```
Comment (Wwwsylvia) [inline Oras.Tests/MemoryTest/MemoryTargetTest.cs:299]: Can we move it to CopyTest? And please have two tests, on for CopyGraph only, and one for Copy.

**Comment by Wwwsylvia** [comment]
I think the  PR title should be "support Copy" or something like that, as you are adding a feature, not just adding tests

File: Oras.Tests/CopyTest.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+namespace Oras.Tests
+{
+    public class CopyTest
+    {
+        /// <summary>
+        /// This method tests if a MemoryTarget object can be copied into another MemoryTarget object
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task MemoryTarget_CanCopyToAnotherMemoryTarget()
```
Comment (Wwwsylvia) [inline Oras.Tests/CopyTest.cs:19]: How about `Copy_MemoryTarget` ?

File: Oras.Tests/CopyTest.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            await sourceTarget.TagAsync(root, reference, cancellationToken);
+            var destinationTarget = new MemoryTarget();
+            var gotDesc = await Copy.CopyAsync(sourceTarget, reference, destinationTarget, "", cancellationToken);
+            Assert.Equal(gotDesc, root);
+
+            foreach (var des in descs)
+            {
+                Assert.True(await destinationTarget.ExistsAsync(des, cancellationToken));
+            }
+        }
```
Comment (Wwwsylvia) [inline Oras.Tests/CopyTest.cs:70]: Need to compare the content as well.

File: Oras.Tests/ContentTest/ContentTest.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Oras.Tests.ContentTest
+{
+    public class ContentTest
+    {
+        /// <summary>
+        /// This method tests if the digest is calculated properly
+        /// </summary>
+        [Fact]
+        public void Content_EnsureThatTheDigestIsCalculatedProperly()
```
Comment (Wwwsylvia) [inline Oras.Tests/ContentTest/ContentTest.cs:17]: `CalculateDigest_xxx`?

**Comment by sammychinedu2ky** [comment]
Test Coverage Report
image

File: Oras.Tests/CopyTest.cs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                await sourceTarget.PushAsync(descs[i], new MemoryStream(blobs[i]), cancellationToken);
+
+            }
+
+            var root = descs[3];
+            var reference = "foobar";
+            await sourceTarget.TagAsync(root, reference, cancellationToken);
+            var destinationTarget = new MemoryTarget();
+            var gotDesc = await Copy.CopyAsync(sourceTarget, reference, destinationTarget, "", cancellationToken);
+            Assert.Equal(gotDesc, root);
```
Comment (Wwwsylvia) [inline Oras.Tests/CopyTest.cs:64]: We should verify `ResolveAsync` as well

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Pull request merged**